FreeRTOS-Plus/Demo/AWS/Fleet_Provisioning_Windows_Simulator/Fleet_Provisioning_With_CSR_Demo/mqtt_operations.c

```diff
@@ -0,0 +1,1069 @@
+                      uint16_t usTopicFilterLength,
+                      const char * pcPayload,
+                      size_t xPayloadLength )
+{
+    bool xReturnStatus = false;
+    MQTTStatus_t xMqttStatus = MQTTSuccess;
+    uint8_t ucPublishIndex = mqttopMAX_OUTGOING_PUBLISHES;
+    MQTTContext_t * pxMqttContext = &xMqttContext;
+
+    configASSERT( pxMqttContext != NULL );
+    configASSERT( pcTopicFilter != NULL );
+    configASSERT( usTopicFilterLength > 0 );
+
+    /* Get the next free index for the outgoing publish. All QoS1 outgoing
+     * publishes are stored until a PUBACK is received. These messages are
+     * stored for supporting a resend if a network connection is broken before
+     * receiving a PUBACK. */
+    xReturnStatus = prvGetNextFreeIndexForOutgoingPublishes( &ucPublishIndex );
+
+    if( xReturnStatus == false )
+    {
+        LogError( ( "Unable to find a free spot for outgoing PUBLISH message." ) );
+    }
+    else
+    {
+        LogDebug( ( "Published payload: %.*s",
+                    ( int ) xPayloadLength,
+                    ( const char * ) pcPayload ) );
+
+        /* This example publishes to only one topic and uses QOS1. */
+        pxOutgoingPublishPackets[ ucPublishIndex ].xPubInfo.qos = MQTTQoS1;
+        pxOutgoingPublishPackets[ ucPublishIndex ].xPubInfo.pTopicName = pcTopicFilter;
+        pxOutgoingPublishPackets[ ucPublishIndex ].xPubInfo.topicNameLength = usTopicFilterLength;
+        pxOutgoingPublishPackets[ ucPublishIndex ].xPubInfo.pPayload = pcPayload;
+        pxOutgoingPublishPackets[ ucPublishIndex ].xPubInfo.payloadLength = xPayloadLength;
+
+        /* Get a new packet id. */
+        pxOutgoingPublishPackets[ ucPublishIndex ].usPacketId = MQTT_GetPacketId( pxMqttContext );
+
+        /* Send PUBLISH packet. */
+        xMqttStatus = MQTT_Publish( pxMqttContext,
+                                    &pxOutgoingPublishPackets[ ucPublishIndex ].xPubInfo,
+                                    pxOutgoingPublishPackets[ ucPublishIndex ].usPacketId );
+
+        if( xMqttStatus != MQTTSuccess )
+        {
+            LogError( ( "Failed to send PUBLISH packet to broker with error = %s.",
+                        MQTT_Status_strerror( xMqttStatus ) ) );
+            prvCleanupOutgoingPublishAt( ucPublishIndex );
+            xReturnStatus = false;
+        }
+        else
+        {
+            LogDebug( ( "PUBLISH sent for topic %.*s to broker with packet ID %u.",
+                        usTopicFilterLength,
+                        pcTopicFilter,
+                        pxOutgoingPublishPackets[ ucPublishIndex ].usPacketId ) );
+        }
+    }
+
+    return xReturnStatus;
+}
+/*-----------------------------------------------------------*/
+
+bool xProcessLoop( void )
+{
+    bool xReturnStatus = false;
+    MQTTStatus_t xMqttStatus = MQTTSuccess;
+
+    xMqttStatus = MQTT_ProcessLoop( &xMqttContext, mqttopMQTT_PROCESS_LOOP_TIMEOUT_MS );
```

Comment:
Reference to a global should be a parameter: &xMqttContext

Reply:
These mqtt_operations changes will be addressed in a future PR, where mqtt_demo_helpers is adapted to work with PKCS11.